### ISSUE TITLE
Unable to press Back button to exit app on root page

### ISSUE THREAD

**Issue by CodingOctocat** [nalu-development/nalu#57]
### Describe the bug

When I use my app for the first time, it will check whether the template database exists or not, if it doesn't exist, the first page will be `CreateTemplatePage`, and then click the `Next` button to enter the Shell, and I use the `OnBackPressedCallback` to confirm the exit, which is working fine in MAUI's standard Shell application. But with Nalu, I can only trigger the `HandleOnBackPressed()` method on `CreateTemplatePage`, and after entering the Shell(NaluShell), pressing the Back button doesn't trigger the `HandleOnBackPressed()`, and I can't even exit the app, nothing happens.

```cs
// .NET9
public partial class App : Application
{
    private readonly Page _firstPage;

    public App(IServiceProvider serviceProvider, INavigationService navigationService)
    {
        InitializeComponent();

        var templateService = serviceProvider.GetRequiredService<TemplateService>();
        var createTemplatePage = serviceProvider.GetRequiredService<CreateTemplatePage>();

#if DEBUG
        _firstPage = createTemplatePage;
#else
        _firstPage = templateService.IsExists()
            ? serviceProvider.GetRequiredService<AppShell>()
            : createFireTemplatePage;
#endif
    }

    protected override Window CreateWindow(IActivationState? activationState)
    {
        return new Window(_firstPage);
    }
}
```

```cs
// @ CreateTemplateViewModel
[RelayCommand]
private void GotoShell()
{
    var shell = _serviceProvider.GetRequiredService<AppShell>(); // AppShell is NaluShell
    Application.Current!.Windows[0].Page = shell;
}
```

```cs
namespace MyProject.Platforms.Android;

[Activity(Theme = "@style/Maui.SplashTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize | ConfigChanges.Density)]
public class MainActivity : MauiAppCompatActivity
{
    private class BackPress : OnBackPressedCallback
    {
        private readonly Activity _activity;

        private long _backPressed;

        public BackPress(Activity activity) : base(true)
        {
            _activity = activity;
        }

        public override void HandleOnBackPressed()
        {
            var navigation = Microsoft.Maui.Controls.Application.Current?.Windows[0]?.Navigation;

            if (navigation is not null && navigation.NavigationStack.Count <= 1 && navigation.ModalStack.Count <= 0)
            {
                const int delay = 2000;

                if (_backPressed + delay > DateTimeOffset.UtcNow.ToUnixTimeMilliseconds())
                {
                    //_activity.FinishAndRemoveTask();
                    //Process.KillProcess(Process.MyPid());
                    _activity.MoveTaskToBack(true);
                }
                else
                {
                    Toast.MakeText(_activity, "Press again to exit", ToastLength.Short)?.Show();
                    _backPressed = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
                }
            }
        }
    }

    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);
        OnBackPressedDispatcher.AddCallback(this, new BackPress(this));
    }
}
```

### Steps to reproduce

as mentioned above

### Expected behaviour

as mentioned above

**Comment by albyrock87** [comment]
@CodingOctocat I would suggest not to use a standalone page as first page, and instead use a `ShellContent` with the flyout menu disabled.
This way you can handle the navigation from the view model and on top of that, the navigation to the main page will also have a smooth transition.

On all your main pages you can override the following method to close the app on back press in Android.


```
protected override bool OnBackButtonPressed() {
    #if ANDROID
    if (Handler?.MauiContext?.Context?.GetActivity() is { } activity) {
        activity.Finish();
    }
    #endif
    return false;
}
```

If you have a fast and synchronous way to know whether to go to the Initialization page or the main page you can do this:

```
public partial class AppShell
{
    public AppShell(
        INavigationService navigationService,
        ICommonService commonService
    )
        : base(navigationService, GetInitialPageType(commonService))
    {
        InitializeComponent();
    }

    private static Type GetInitialPageType(ICommonService commonService)
    {
        if (commonService.NeedsInitialization()))
        {
            return typeof(InitPage);
        }

        return typeof(MainPage);
    }
}
```

Let me know if this solves your issue or if you have further questions/suggestions.

**Comment by CodingOctocat** [comment]
But even if I don't have a `CreateTemplatePage` and go directly to the Shell, I can't press Back to exit (I can do it in a subpage, but not in the Shell Root), and it seems that NaluShell doesn't respond to the Back button if NavigationStack is empty.

```cs
public partial class App : Application
{
    private readonly Page _firstPage;

    public App(IServiceProvider serviceProvider, INavigationService navigationService)
    {
        InitializeComponent();
        _firstPage = serviceProvider.GetRequiredService<AppShell>();
    }

    protected override Window CreateWindow(IActivationState? activationState)
    {
        return new Window(_firstPage);
    }
}
```

**Comment by albyrock87** [comment]
@CodingOctocat your issue should be solved with version 8.2.7, please let me know if it works now.

That said, I still recommend the approach of having `AppShell` as main page from the beginning.

**Comment by CodingOctocat** [comment]
@albyrock87 It work fine! Thanks!
---
For your recommended method, does that mean using AppShell as the firstPage and overlaying it with a Modal page?
